openwisp_controller/config/tests/test_template.py

```diff
@@ -695,3 +696,30 @@ def test_task_timeout(self, mocked_update_related_config_status):
             template.save()
             mocked_error.assert_called_once()
         mocked_update_related_config_status.assert_called_once()
+
+    def test_required_vpn_client_deleted(self):
+        org = self._get_org()
+        vpn = self._create_vpn()
+        template = self._create_template(
+            name='vpn-test', type='vpn', vpn=vpn, auto_cert=True,
+        )
+        config = self._create_config(organization=org)
+        config.templates.add(template)
+        vpn_client = config.vpnclient_set.first()
+        self.assertIsNotNone(vpn_client)
+
+        db = router.db_for_write(vpn_client._meta.model, instance=config)
+        template.delete()
+        vpn_client.refresh_from_db()
+        self.assertEqual(config.templates.count(), 0)
```

Comment:
why are you deleting the template?
This is confusing.

Reply:
Deleting the template does not trigger m2m_changed with `custom_openwisp` attribute. This helped in testing that the required operation is performed by manually sending m2m_changed signal. 

This test will not be required once we add tests for django admin